src/electos/ballotmaker/ballots/contest_layout.py

```diff
@@ -256,30 +262,48 @@ def __init__(self, contest_data: CandidateContestData):
             self.instruct = f"Vote for up to {self.votes_allowed}"
         else:
             self.instruct = f"Vote for {self.votes_allowed}"
-        self.candidates = contest_data.candidates
+        logging.info(f"Candidate contest: {self.title}")
+        self.candidate_choices = contest_data.candidates
         _selections = []
 
-        for candidate in self.candidates:
-            # add newlines around " and "
-            if candidate.name.find(" and "):
-                candidate.name = candidate.name.replace(
-                    " and ", "<br />and<br />"
-                )
-            # make the candidate name bold
-            contest_text = f"<b>{candidate.name}</b>"
-            # add party abbreviation in plain text
-            if candidate.party_abbr != "":
-                contest_text += f"<br />{candidate.party_abbr}"
-            contest_object = [Paragraph(contest_text, normal)]
+        for candidate_choice in self.candidate_choices:
+
             # add line for write ins
-            if candidate.is_write_in:
-                # contest_text += ("<br />" * 2) + ("_" * 20)
+            if candidate_choice.is_write_in:
+                logging.info(
+                    f"Found write-in candidate: {candidate_choice.id}"
+                )
+                contest_text = "<b>or write in:</b>"
+                # contest_text = ("<br />" * 2) + ("_" * 20)
+                contest_object = [Paragraph(contest_text, normal)]
                 # Add text field with ID and suffix
-                input_id = f"{candidate.id}_text"
+                input_id = f"{candidate_choice.id}_text"
                 contest_object.append(formInputField(input_id))
+            else:
+                contest_text = ""
+                for candidate_count, name in enumerate(
+                    candidate_choice.name, start=1
+                ):
+                    if candidate_count > 1:
+                        contest_text += "<br />and<br />"
+                    # make the candidate name bold
+                    contest_text += f"<b>{name}</b>"
+                # add party abbreviation in plain text
+                party_count = len(candidate_choice.party)
+                if party_count == 1:
+                    contest_text += (
+                        f"<br />{candidate_choice.party[0].abbreviation}"
+                    )
+                elif party_count > 1:
+                    raise NotImplementedError(
+                        "Multiple party tickets not supported (parties"
```

Comment:
Missing a close `)` or the rest of a sentence.

Reply:
Unfinished code! Now reads:

```
                elif party_count > 1:
                    raise NotImplementedError(
                        f"Multiple party tickets not supported (parties found: {party_count})"
```